lib/rules/id-length.js

```diff
@@ -63,8 +70,19 @@ module.exports = {
 
                 return obj;
             }, {});
+        const exceptionPatterns = (options.exceptionPatterns || []).map(pattern => new RegExp(pattern, "u"));
         const reportedNode = new Set();
 
+        /**
+         * Checks if a string matches the provided exception patterns
+         * @param {string} name The string to check.
+         * @returns {boolean} if the string is a match
+         * @private
+         */
+        function matchesExceptions(name) {
```

Comment:
The name of this function might be confusing since we also have the `exceptions` array, so it isn't clear from its name whether this function checks `exceptions`, or `exceptionPatterns`, or maybe both.

Maybe we could rename this function to `matchesExceptionPattern`?

Reply:
You are right. It will be better to avoid confusion using that. Thanks for the advice!